### PR TITLE
WIP: Added a method for downloading the users as a CSV

### LINKED ISSUE #658: .csv export of vendor data
As an admin user, I will sometimes want to get an export of vendor data so I can do things like send bulk emails to the vendors. So the system should allow me to do this

Context:
Given I am a logged in admin user

Scenario: I see a link on the users page to dowload a .csv
When I am viewing the users page https://micropurchase.18f.gov/admin/users
Then I see a link labeled "Download CSV" next to the "Users" header

Scenario: I download a .csv containing users name and email
When I click on the "Download CSV" link
Then a CSV file should be downloaded to my computer
And the CSV column headers should be "First Name", "Last Name" and "Email Address"
And under each column header I should see the first name, last name and email address of each user

### PR DESCRIPTION
**Pull request by harrisj**
Fixes #658

### CODE REVIEW

File: spec/controllers/admin/user_reports_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe Admin::UserReportsController do
+  describe '#index' do
+    context 'when called by an admin user' do
+      it 'should return a CSV' do
+        user = create(:admin_user)
+        10.times { create(:user) }
```
Comment (jessieay) [inline spec/controllers/admin/user_reports_controller_spec.rb]: do we need any users in the system in order for this csv export to work?

Comment (harrisj) [inline spec/controllers/admin/user_reports_controller_spec.rb]: It should work without I think

File: spec/factories/users.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@
     duns_number { Faker::Company.duns_number }
     name { Faker::Name.name }
     email
+    github_login 'ghlogin'
```
Comment (jessieay) [inline spec/factories/users.rb]: maybe this should be a required field on `User` ?

File: app/controllers/admin/user_reports_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class Admin::UserReportsController < ApplicationController
+  def index
```
Comment (jessieay) [inline app/controllers/admin/user_reports_controller.rb]: do we want to ensure that only admins can access this endpoint?

File: spec/controllers/admin/user_reports_controller_spec.rb (hunk 1 above)
Comment (jessieay) [inline spec/controllers/admin/user_reports_controller_spec.rb]: in that case, thoughts on not creating these records? would make tests faster ;)

**Comment by jessieay** [comment]
Few small comments -- looking awesome!

**Comment by jessieay** [comment]
looks like the story contains a requirement of a link to download this CSV -- are you planning on doing that separately?

**Comment by harrisj** [comment]
Good point. I should add that

File: spec/controllers/admin/user_reports_controller_spec.rb (hunk 1 above)
Comment (harrisj) [inline spec/controllers/admin/user_reports_controller_spec.rb]: I wanted to have at least 2 for kicks... but yeah

File: spec/factories/users.rb (hunk 2 above)
Comment (harrisj) [inline spec/factories/users.rb]: I actually need to check if we are fetching this anywhere or not

**Comment by harrisj** [comment]
I added the link, but it could use some styling assistance.

File: spec/controllers/admin/user_reports_controller_spec.rb (hunk 1 above)
Comment (jessieay) [inline spec/controllers/admin/user_reports_controller_spec.rb]: cool can we just create 2 additional users, then? fast test suite = more helpful test suite ;)

File: spec/controllers/admin/user_reports_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+        get :index, { format: 'csv' }, user_id: user.id
+
+        expect(response.code).to eq('200')
+        expect(response.header['Content-Type']).to include('text/csv')
+      end
+    end
+
+    context 'when called by a non-admin user' do
+      it 'should return a 302 redirection with an error' do
```
Comment (jessieay) [inline spec/controllers/admin/user_reports_controller_spec.rb]: niiiice

File: spec/factories/users.rb (hunk 2 above)
Comment (jessieay) [inline spec/factories/users.rb]: why did we add it here? If there is not a validation but we require it for something to work, we should add the validation

Comment (jessieay) [inline spec/factories/users.rb]: fwiw, anyone signing in via github will have a github login so adding the validation shouldn't be a problem

Comment (harrisj) [inline spec/factories/users.rb]: We wanted to grab it when names weren't being captured.

**Comment by jessieay** [comment]
this looks perfecto -- merging!

**Pull request merged**